variables.tf

```diff
@@ -0,0 +1,80 @@
+/*
+Module variables
+*/
+
+variable "alb_is_internal" {
+  description = "Determines if the ALB is internal. Default: false"
+  default     = false
+}
+
+variable "alb_name" {
+  description = "The name of the ALB as will show in the AWS EC2 ELB console."
+  default     = "my-alb"
+}
+
+variable "alb_protocols" {
+  description = "A comma delimited list of the protocols the ALB accepts. e.g.: HTTPS"
+  default     = "HTTPS"
+}
+
+variable "alb_security_groups" {
+  description = "A comma separated string of security groups with which we associate the ALB. e.g. 'sg-edcd9784,sg-edcd9785'"
+  type        = "list"
+}
+
+variable "aws_region" {
+  description = "AWS region to use."
+}
+
+variable "backend_port" {
+  description = "The port the service on the EC2 instances listen on."
+  default     = 80
```

Comment:
No assumption. Make it fail, if user put bad value.

Reply:
Again here, providing a default of what many if not most people would use (and a collection of defaults that work together) is not a bad approach. Following "make no assumptions " to a logical conclusion leaves a module consumer requiring to specify every last variable which strips some of the value of modules. Modules are intended to enable people to get started quickly with bundled resources. Part of that is adding defaults that are likely to fit many use cases.